src/Avalonia.Base/Media/TextFormatting/TextEllipsisHelper.cs

```diff
@@ -37,7 +37,7 @@ internal static class TextEllipsisHelper
                     {
                         case ShapedTextRun shapedRun:
                             {
-                                currentWidth += shapedRun.Size.Width;
+                                currentWidth = shapedRun.Size.Width;
```

Comment:
`currentWidth` can be removed here because we do not update its value locally. We can use `shapedRun.Size.Width` directly. This makes it clearer.